### PR TITLE
refactor ion Tag

### PR DESCRIPTION
**Pull request by lucasrasec**
Removing icon requirement in component

Adding empty label treatment

Update in validateHexColor function

### CODE REVIEW

File: projects/ion/src/lib/tag/tag.component.spec.ts [hunk 1]
```diff
@@ -98,6 +98,25 @@ describe('TagComponent', () => {
...
+    await sut({ ...defaultValue, icon: '' });
+    expect(screen.queryAllByTestId(IDs.icon)).toHaveLength(0);
+  });
+
+  it('should not render component and throw error when label is empty', async () => {
+    try {
+      await sut({ label: '' });
+    } catch (e) {
+      expect(screen.queryAllByTestId(IDs.tag)).toHaveLength(0);
+      expect(e.message).toBe('Invalid Tag label informed');
```
Comment (iurynogueira) [inline projects/ion/src/lib/tag/tag.component.spec.ts:116]:
```suggestion
      expect(error.message).toBe('Invalid Tag label informed');
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]

File: projects/ion/src/lib/tag/tag.component.ts [hunk 2]
```diff
@@ -38,4 +38,18 @@ export class TagComponent {
   getTagColor() {
     return validateHexColor(this.color) ? this.color : defaultColor;
   }
+
+  hasLabel() {
+    return !this.label || this.label.trim() === '' ? true : false;
```
Comment (iurynogueira) [inline projects/ion/src/lib/tag/tag.component.ts:43]:
```suggestion
    return !this.label || this.label.trim() === '';
```

**Review by iurynogueira — CHANGES_REQUESTED** [review]

File: projects/ion/src/lib/tag/tag.component.ts [hunk 3]
```diff
@@ -38,4 +38,18 @@ export class TagComponent {
...
+  }
+
+  validateLabel() {
+    if (this.hasLabel()) {
+      throw new Error('Invalid Tag label informed');
+    }
+  }
+
+  ngOnInit(): void {
+    this.validateLabel();
```
Comment (mizek1) [inline projects/ion/src/lib/tag/tag.component.ts:53]: Hmmm I think we have a problem putting this validation only at onInit().
If I render <ion-tag label="someVariable"> with someVariable = 'label', and then change someVariable to "        ", tag would not emit any error and will appear empty. ([Here you can see a demonstration of what I'm talking about.](https://user-images.githubusercontent.com/57760325/191968780-72cd6d7e-7931-4428-8dba-0eb076bc72cc.webm))
You can validate with @edneyql or @iurynogueira if this is really a problem or no.

Comment (mizek1) [reply projects/ion/src/lib/tag/tag.component.ts:53]: I think this can be solved with ngOnChanges.

Comment (iurynogueira) [reply projects/ion/src/lib/tag/tag.component.ts:53]: I think that a problem

**Review by mizek1 — APPROVED** [review]

**Pull request merged**